apps/web/src/components/EmailVerification/EmailVerificationDialog.tsx

```diff
@@ -0,0 +1,540 @@
+import { IntlShape, useIntl } from "react-intl";
+import {
+  FormProvider,
+  SubmitHandler,
+  useForm,
+  useWatch,
+} from "react-hook-form";
+import { useMutation } from "urql";
+
+import { Button, Dialog, Well } from "@gc-digital-talent/ui";
+import { Input, Submit } from "@gc-digital-talent/forms";
+import { errorMessages, commonMessages } from "@gc-digital-talent/i18n";
+import { toast } from "@gc-digital-talent/toast";
+import { EmailType, graphql } from "@gc-digital-talent/graphql";
+import {
+  assertUnreachable,
+  emptyToNull,
+  notEmpty,
+  workEmailDomainRegex,
+} from "@gc-digital-talent/helpers";
+
+const CODE_REQUEST_THROTTLE_DELAY_S = 60;
+
+const EmailVerificationRequestACode_Mutation = graphql(/* GraphQL */ `
+  mutation EmailVerificationRequestACode(
+    $input: SendUserEmailsVerificationInput!
+  ) {
+    sendUserEmailsVerification(sendUserEmailsVerificationInput: $input) {
+      id
+    }
+  }
+`);
+
+const EmailVerificationSubmitACode_Mutation = graphql(/* GraphQL */ `
+  mutation EmailVerificationSubmitACode($code: String!) {
+    verifyUserEmails(code: $code) {
+      id
+    }
+  }
+`);
+
+const getDescription = (
+  emailType: EmailVerificationProps["emailType"],
+  intl: IntlShape,
+) => {
+  switch (emailType) {
+    case EmailType.Work:
+      return intl.formatMessage({
+        defaultMessage:
+          "In order to verify your work email, the domain must match a known Government of Canada email pattern (e.g. @canada.ca, @department.gc.ca, etc.).",
+        id: "RiaRHW",
+        description: "Work email title paragraph",
+      });
+    case EmailType.Contact:
+    default:
+      return intl.formatMessage({
+        defaultMessage:
+          "This email will be used by recruitment and HR teams to contact you about opportunities and will also receive notifications about your applications and other platform details.",
+        id: "Dd8fch",
+        description: "Contact email title paragraph",
+      });
+  }
+};
+
+const getLabel = (
+  emailType: EmailVerificationProps["emailType"],
+  intl: IntlShape,
+) => {
+  switch (emailType) {
+    case EmailType.Work:
+      return intl.formatMessage(commonMessages.workEmail);
+    case EmailType.Contact:
+    default:
+      return intl.formatMessage(commonMessages.email);
+  }
+};
+
+const getSubtitle = (
+  emailType: EmailVerificationProps["emailType"],
+  intl: IntlShape,
+) => {
+  switch (emailType) {
+    case EmailType.Work:
+      return intl.formatMessage({
+        defaultMessage:
+          "Verify your Government of Canada work email to confirm your status as an employee.",
+        id: "UoTwQ8",
+        description: "Work email subtitle",
+      });
+    case EmailType.Contact:
+    default:
+      return intl.formatMessage({
+        defaultMessage:
+          "Add and verify a contact email that will be used for communication and notifications.",
+        id: "upuqPk",
+        description: "Contact email subtitle",
+      });
+  }
+};
```

Comment:
Can these all be combined since they all repeat the same swtich case structure?

You could also just do something like 

```tsx
let description;
let label;
let subtitle;

if (emailType === EmailType.Work) {
  description = intl.formatMessage();
  label = intl.formatMessage();
  subtitle = intl.formatMessage();
} else {
  description = intl.formatMessage();
  label = intl.formatMessage();
  subtitle = intl.formatMessage();
}

```

Reply:
Subjective, but I usually prefer smaller functions with a single purpose that can be composed.  It is kind of ugly, though.  I refactored this to not need functions or case statements at all.  What do you think?
[refactor to message maps](https://github.com/GCTC-NTGC/gc-digital-talent/pull/14449/commits/c4eaf6d591bca9ec59d447c21548b62112d24566)